drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c

```diff
@@ -258,26 +263,44 @@ static inline int ieee802154_cc13xx_cc26xx_subg_channel_to_frequency(
 	__ASSERT_NO_MSG(frequency != NULL);
 	__ASSERT_NO_MSG(fractFreq != NULL);
 
+	/* See IEEE 802.15.4, section 10.1.3.3. */
 	if (channel == IEEE802154_SUB_GHZ_CHANNEL_MIN) {
 		*frequency = 868;
 		/*
 		 * uint16_t fractional part of 868.3 MHz
 		 * equivalent to (0.3 * 1000 * BIT(16)) / 1000, rounded up
 		 */
 		*fractFreq = 0x4ccd;
-	} else if (1 <= channel && channel <= IEEE802154_SUB_GHZ_CHANNEL_MAX) {
+	} else if (channel <= IEEE802154_SUB_GHZ_CHANNEL_MAX) {
 		*frequency = 906 + 2 * (channel - 1);
 		*fractFreq = 0;
-	} else if (IEEE802154_2_4_GHZ_CHANNEL_MIN <= channel
-		&& channel <= IEEE802154_2_4_GHZ_CHANNEL_MAX) {
-		*frequency = 2405 + 5 * (channel - IEEE802154_2_4_GHZ_CHANNEL_MIN);
-		*fractFreq = 0;
 	} else {
 		*frequency = 0;
 		*fractFreq = 0;
 		return -EINVAL;
 	}
 
+	/* TODO: This incorrectly mixes up legacy O-QPSK SubGHz PHY channel page zero
```

Comment:
IIRC, this was the only way to make it work with wpanusb.

Reply:
The wpanusb code seems to be limited to channel page zero but then uses the channel number abstraction so it should be independent of low-level PHY settings (including but not limited to the modulation scheme). The original code by 
[finikorg](https://github.com/finikorg/wpanusb) has a hard coded channel list, [@jadonk's fork](https://github.com/jadonk/wpanusb) fixes that for the supported channel list but still seems to have channel page zero hard coded.

wpanusb allows for a channel page being set in set_channel() but it doesn't look as if this also updated the supported channels, so probably setting a non-zero channel page will break the supported channels which again will let the kernel driver bail out if trying to set a channel above 10 (which is totally valid on channel page 9/10 for example). This means that probably wpanusb will not work with channel pages 9 or 10 (SUN FSK) but continues to work with channel page zero (which must not be removed anyway).

The problem is that CC13/26xx doesn't support channel page zero in the subG area at all as it doesn't support O-QPSK subG modulation. Therefore there's no way to fix the PHY for this driver for channel page zero. This means that "fixing" in this case only means additionally supporting channel page 9 but leaving channel page zero exactly as it is (including its non-standard PHY).

Hope that explains a bit what that change is going to be about?